s3_upload/utils/upload.py

```
@@ -399,36 +399,45 @@
+            all_uploaded_files = {**all_uploaded_files, **uploaded_files}
+            all_failed_upload.extend(failed_upload)
+        except Exception as exc:
+            # catch any other errors that might get raised from the
+            # ProcessPool but not handled in the child ThreadPool
+            all_failed_upload.extend(concurrent_jobs[future])
+            log.error(
+                "Failed uploading %s files for a single ProcessPool due to"
+                " an unhandled error: %s",
+                len(concurrent_jobs[future]),
+                exc,
+            )
```

Ruff (0.7.0)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)